pyaedt/desktop.py

```diff
@@ -363,203 +366,193 @@ def current_version_student(self):
-                        output = tasks_list.readlines()
-                    pattern = r"(?i)^(?:{})\s+?(\d+)\s+.+[\s|\\](?:{})\s+".format(process, username)
-                    for l in output:
-                        m = re.search(pattern, l)
-                        if m:
-                            processID2.append(m.group(1))
-
-                proc = [i for i in processID2 if i not in processID]
-                if not proc:
-                    if NG:
-                        self._main.close_on_exit = False
-                    else:
-                        self._main.close_on_exit = False
-                    oAnsoftApp = win32com.client.Dispatch(version)
-                    self._main.oDesktop = oAnsoftApp.GetAppDesktop()
-                    self._main.isoutsideDesktop = True
-                elif version_key >= "2021.1":
-                    self._main.close_on_exit = True
-                    module_logger.debug("Info: {} Started with Process ID {}".format(version, proc[0]))
-                    context = pythoncom.CreateBindCtx(0)
-                    running_coms = pythoncom.GetRunningObjectTable()
-                    monikiers = running_coms.EnumRunning()
-                    for monikier in monikiers:
-                        m = re.search(
-                            version[10:] + r"\.\d:" + str(proc[0]), monikier.GetDisplayName(context, monikier)
-                        )
-                        if m:
-                            obj = running_coms.GetObject(monikier)
-                            self._main.isoutsideDesktop = True
-                            # self._main.oDesktop = win32com.client.gencache.EnsureDispatch(
-                            #     obj.QueryInterface(pythoncom.IID_IDispatch))
-                            self._main.oDesktop = win32com.client.Dispatch(obj.QueryInterface(pythoncom.IID_IDispatch))
-                            break
-                else:
-                    module_logger.warning("PyAEDT is not supported in AEDT versions older than 2021.1.")
-                    oAnsoftApp = win32com.client.Dispatch(version)
-                    self._main.oDesktop = oAnsoftApp.GetAppDesktop()
-                    self._main.isoutsideDesktop = True
+            version_student, version_key, version = self._set_version(specified_version, student_version)
+            if _com == 'ironpython':
+                print("Launching PyAEDT outside Electronics Desktop with IronPython")
+                self._init_ironpython(NG, AlwaysNew)
+            elif _com == 'pythonnet_v3':
+                print("Launching PyAEDT outside Electronics Desktop with CPython and Pythonnet")
+                self._init_cpython(NG, AlwaysNew, version, student_version, version_key)
             else:
-                module_logger.debug("Launching AEDT with Module win32com.client")
+                print("Launching PyAEDT outside Electronics Desktop with CPython and PyWin32")
```

Comment:
We really need to avoid printing.

Reply:
```suggestion
                self.add_info_message("Launching PyAEDT outside AEDT with CPython and PyWin32.")
```
Combining suggestions from both @akaszynski  and @PipKat  here.